### PR TITLE
Output resources in azure.yaml after app init

### PR DESCRIPTION
**Pull request by saragluna**
This PR will output the resources detected from app init to `azure.yaml`.

### CODE REVIEW

File: cli/azd/pkg/project/resources.go [hunk 1]
```diff
@@ -79,6 +82,11 @@ func (r *ResourceConfig) MarshalYAML() (interface{}, error) {
 		if err != nil {
 			return nil, err
 		}
+	case ResourceTypeDbMySQL:
+		err := marshalRawProps(raw.Props.(MySQLProps))
+		if err != nil {
+			return nil, err
+		}
```
Comment (rujche) [inline cli/azd/pkg/project/resources.go:89]: How about other resources: PostgreSQL, MnogoDB, Redis?

Comment (rujche) [inline cli/azd/pkg/project/resources.go:89]: Is it necessary to add other resource type like: ResourceTypeDbRedis, ResourceTypeDbPostgres, ResourceTypeDbMong?

File: cli/azd/pkg/project/resources.go [hunk 2]
```diff
@@ -118,6 +126,12 @@ func (r *ResourceConfig) UnmarshalYAML(value *yaml.Node) error {
 			return err
 		}
 		raw.Props = cap
+	case ResourceTypeDbMySQL:
+		mp := MySQLProps{}
+		if err := unmarshalProps(&mp); err != nil {
+			return err
+		}
+		raw.Props = mp
```
Comment (rujche) [inline cli/azd/pkg/project/resources.go:134]: Same here: Is it necessary to add other resource type like: ResourceTypeDbRedis, ResourceTypeDbPostgres, ResourceTypeDbMong?

File: cli/azd/pkg/project/resources.go (hunk 1 above)
Comment (rujche) [reply cli/azd/pkg/project/resources.go:89]: Could you please add unit test for `MarshalYAML` and `UnmarshalYAML`?

**Review by haoozhang — APPROVED** [review]

**Pull request merged**